core/src/main/java/com/rudderstack/android/sdk/core/FlushUtils.java

```diff
@@ -0,0 +1,244 @@
+package com.rudderstack.android.sdk.core;
+
+import android.text.TextUtils;
+
+import androidx.annotation.Nullable;
+
+import com.rudderstack.android.sdk.core.util.Utils;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Standard flush related calls
+ */
+class FlushUtils {
+    private FlushUtils() {
+
+    }
+
+    private static final Object FLUSH_LOCK = new Object();
+    private static final Object DB_LOCK = new Object();
+
+    /**
+     * Should not be called from main thread.
+     *
+     * @param areFactoriesInitialized
+     * @param integrationOperationsMap
+     * @param messageIds
+     * @param messages
+     * @param flushQueSize
+     * @param dataPlaneUrl
+     * @param dbManager
+     * @param authHeaderString
+     * @param anonymousIdHeaderString
+     * @return
+     */
+    static boolean flush(boolean areFactoriesInitialized, @Nullable Map<String, RudderIntegration<?>> integrationOperationsMap,
+                         List<Integer> messageIds, List<String> messages, int flushQueSize, String dataPlaneUrl,
+                         DBPersistentManager dbManager,
+                         String authHeaderString, String anonymousIdHeaderString) {
+
+        if (areFactoriesInitialized && integrationOperationsMap != null) {
+            RudderLogger.logDebug("EventRepository: flush native SDKs");
+            for (String key : integrationOperationsMap.keySet()) {
+                RudderLogger.logDebug(String.format(Locale.US, "EventRepository: flush for %s", key));
+                RudderIntegration<?> integration = integrationOperationsMap.get(key);
+                if (integration != null) {
+                    integration.flush();
+                }
+            }
+        }
+        Utils.NetworkResponses networkResponse;
+
+        messageIds.clear();
```

Comment:
Not sure I understood this, why do we get messageIds as argument and then clear it?

Reply:
These are legacy codes for fetching and uploading, It seems from the code is that messages and messageIds, that were sent as params, are expected to contain the unsent data.
The parameters are expected to be empty lists, which will be filled with data in this method. This is to ensure that it's empty.